providers/github/resources/github_org.go

```diff
@@ -262,26 +265,66 @@ func (g *mqlGithubOrganization) repositories() ([]interface{}, error) {
 		return nil, g.Login.Error
 	}
 	orgLogin := g.Login.Data
-
-	listOpts := &github.RepositoryListByOrgOptions{
-		ListOptions: github.ListOptions{PerPage: paginationPerPage},
-		Type:        "all",
+	listOpts := github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{
+			PerPage: paginationPerPage,
+			Page:    1,
+		},
+		Type: "all",
 	}
 
+	repoCount := g.TotalPrivateRepos.Data + g.TotalPublicRepos.Data
+	workerPool := workerpool.New[[]*github.Repository](workers)
+	workerPool.Start()
+	defer workerPool.Close()
+
+	log.Debug().
+		Int("workers", workers).
+		Int64("total_repos", repoCount).
+		Str("organization", g.Name.Data).
+		Msg("list repositories")
+
 	var allRepos []*github.Repository
 	for {
-		repos, resp, err := conn.Client().Repositories.ListByOrg(conn.Context(), orgLogin, listOpts)
-		if err != nil {
+
+		// exit as soon as we collect all repositories
+		if len(allRepos) >= int(repoCount) {
+			break
+		}
+
+		// send as many request as workers we have
```

Comment:
Why do we limit by number of workers? Doesn't this artificially throttle things

Reply:
Good point. Let me check that.

Reply:
@jaym I updated the PR